### PR TITLE
CLN: resolve UserWarning in `pandas/plotting/_matplotlib/core.py` #35945

### LINKED ISSUE #35945: CLN: UserWarning in `pandas/plotting/_matplotlib/core.py`
```
pandas/tests/plotting/test_frame.py: 10 warnings
pandas/tests/plotting/test_series.py: 5 warnings
  /home/fangchenli/Workspace/pandas-VirosaLi/pandas/plotting/_matplotlib/core.py:1235: UserWarning: FixedFormatter should only be used together with FixedLocator
    ax.set_xticklabels(xticklabels)
```

### PR DESCRIPTION
**Pull request by fangchenli**
- [x] closes #35945
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by charlesdong1991 — CHANGES_REQUESTED** [review]
thanks for the PR, @VirosaLi 

just curious where is this warning getting caught or is the warning raised on a newer matplotlib version, do we have any tests right now to catch the warning?

**Comment by VirosaLi** [comment]
> thanks for the PR, @VirosaLi
> 
> just curious where is this warning getting caught or is the warning raised on a newer matplotlib version, do we have any tests right now to catch the warning?

pandas/tests/plotting/test_frame.py: 10 warnings
pandas/tests/plotting/test_series.py: 5 warnings
  /home/fangchenli/Workspace/pandas-VirosaLi/pandas/plotting/_matplotlib/core.py:1235: UserWarning: FixedFormatter should only be used together with FixedLocator
    ax.set_xticklabels(xticklabels)

File: pandas/plotting/_matplotlib/core.py [hunk 1]
```diff
@@ -1232,8 +1230,8 @@ def get_label(i):
         if self._need_to_set_index:
             xticks = ax.get_xticks()
             xticklabels = [get_label(x) for x in xticks]
+            ax.set_xticks(xticks)
```
[the remark this answers is not in the document]
Comment (VirosaLi) [reply pandas/plotting/_matplotlib/core.py:1233]: I agree with you. Let me take another look.

**Comment by jreback** [comment]
@VirosaLi so the warning is now gone? I don't think we actually need a test for this, but if you can put an assert_produces_warning(None) around something and have it pass would be great.

**Review by charlesdong1991 — APPROVED** [review]
lgtm

if you could get the CI green (failure doesn't seem relevant to the PR though, maybe try to rebase one more time)

**Comment by charlesdong1991** [comment]
the first one will be resolved once https://github.com/pandas-dev/pandas/pull/36085 gets merged.

**Comment by jreback** [comment]
thanks @VirosaLi

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x
